### PR TITLE
docs(CHANGELOG): update 3.4/3.5 changelogs with defrag fixes

### PR DESCRIPTION
**Pull request by ghouscht**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: CHANGELOG/CHANGELOG-3.4.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Previous change logs can be found at [CHANGELOG-3.3](https://github.com/etcd-io/
 
 ### etcd server
 - Fix [watchserver related goroutine leakage](https://github.com/etcd-io/etcd/pull/18785)
+- Fix [handle errors during defrag](https://github.com/etcd-io/etcd/pull/18843)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.4.md:11]:
```suggestion
- Fix [panicking occurred due to improper error handling during defragmentation](https://github.com/etcd-io/etcd/pull/18843)
```

File: CHANGELOG/CHANGELOG-3.5.md [hunk 2]
```diff
@@ -8,6 +8,7 @@ Previous change logs can be found at [CHANGELOG-3.4](https://github.com/etcd-io/
 ### etcd server
 - Fix [watchserver related goroutine leakage](https://github.com/etcd-io/etcd/pull/18784)
 - Fix [risk of a partial write txn being applied](https://github.com/etcd-io/etcd/pull/18799)
+- Fix [handle errors during defrag](https://github.com/etcd-io/etcd/pull/18842)
```
Comment (ahrtr) [inline CHANGELOG/CHANGELOG-3.5.md:11]: ditto

**Comment by ivanvc** [comment]
Link to #18810

**Review by ivanvc — APPROVED** [review]
LGTM on green. Thanks, @ghouscht.

**Review by ahrtr — APPROVED** [review]

**Pull request merged**